aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1511,7 +1511,12 @@ impl AptosVM {
         new_published_modules_loaded: &mut bool,
         change_set_configs: &ChangeSetConfigs,
     ) -> Result<UserSessionChangeSet, VMStatus> {
-        let maybe_publish_request = session.execute(|session| session.extract_publish_request());
+        let maybe_publish_request = session.execute(|session| {
+            // With V2 loader, do not allow init_module to request module publish. While allowing
+            // it is harmless, the user experience can be bad - so we better avoid it completely.
```

Comment:
nit: let's replace EX can be bad with specific example, e.g. confusing, etc. unless it's needed for some use case in which case one can think about the best UX (but I doubt it would ever be needed)

Reply:
Added a new feature flag, and a comment why this is used.